bin/propolis-server/src/lib/server.rs

```diff
@@ -1144,9 +1170,17 @@ async fn instance_issue_crucible_vcr_request(
     // Crucible does the heavy lifting here to verify that the old/new
     // VCRs are different in just the correct way and will return error
     // if there is any mismatch.
-    backend.vcr_replace(old_vcr_json, &new_vcr_json).await.map_err(|e| {
-        HttpError::for_bad_request(Some(e.to_string()), e.to_string())
-    })?;
+    let replace_result =
+        match backend.vcr_replace(old_vcr_json, &new_vcr_json).await {
+            Ok(replace_result) => replace_result,
+
+            Err(e) => {
+                return Err(HttpError::for_bad_request(
+                    Some(e.to_string()),
+                    e.to_string(),
+                ));
+            }
+        };
```

Comment:
Why the explicit return vs the `?` operator (like above)?

Reply:
Sorry, do you mean why doesn't it look like

```rust
    let replace_result =
        backend.vcr_replace(old_vcr_json, &new_vcr_json).await.map_err(
            |e| HttpError::for_bad_request(Some(e.to_string()), e.to_string()),
        )?;
```

? I could change it to that, yeah.

Reply:
cool, done in eb21550